### PR TITLE
Feature: removes the amazon-ssm-agent 'snap'

### PR DESCRIPTION
**Pull request by lsh-0**
and purges snapd, the snap daemon

### CODE REVIEW

**Comment by lsh-0** [comment]
apparently if you don't remove the 'snap' files and cache dir, then apt can't purge properly

**Comment by lsh-0** [comment]
looks like reverse requisites aren't being obeyed. again. fml

File: elife/base.sls [hunk 1]
```diff
@@ -71,10 +74,28 @@ ubuntu-user:
...
             - hash snap
+        - require:
+            - amazon-ssm-agent-snap-removal
+
+    cmd.run:
+        - name: rm -rf /var/cache/snapd
+        - require:
+            - service: snapd
+        - require_in:
+            - base-purging
```
Comment (giorgiosironi) [inline elife/base.sls]: lore says that you need to specify both the state type and its name in reverse requisites, `cmd: base-purging` (or whatever it is)

Comment (lsh-0) [reply elife/base.sls]: ah - I'll give that a shot, cheers.

Comment (lsh-0) [reply elife/base.sls]: whaddya know, it's green

**Pull request merged**